core/MatomoInstaller.php

```diff
@@ -0,0 +1,416 @@
+
+        if (is_null($settings['port']))
+        {
+            $port = Adapter::getDefaultPortForAdapter($settings['adapter']);
+            $settings['port'] = $port;
+        }
+
+        //DEFAULT TABLES_PREFIX could be define in global.ini.php
+        //(Could table be without prefix ?)
+        if (is_null($settings['tables_prefix']))
+            $settings['tables_prefix'] = "matomo_";
+
+        //Prepare db settings.
+        $dbInfos = array(
+            'host'          => $settings['dbhost'],
+            'username'      => $settings['dbusername'],
+            'password'      => $settings['dbpassword'],
+            'dbname'        => $settings['dbname'],
+            'tables_prefix' => $settings['tables_prefix'],
+            'adapter'       => $settings['adapter'],
+            'port'          => $settings['port'],
+            'schema'        => Config::getInstance()->database['schema'],
+            'type'          => $settings['type'],
+            'enable_ssl'    => false
+        );
+
+        if (($portIndex = strpos($dbInfos['host'], '/')) !== false) {
+            // unix_socket=/path/sock.n
+            $dbInfos['port'] = substr($dbInfos['host'], $portIndex);
+            $dbInfos['host'] = '';
+        } else if (($portIndex = strpos($dbInfos['host'], ':')) !== false) {
+            // host:port
+            $dbInfos['port'] = substr($dbInfos['host'], $portIndex + 1);
+            $dbInfos['host'] = substr($dbInfos['host'], 0, $portIndex);
+        }
+
+        try {
+            @Db::createDatabaseObject($dbInfos);
+        } catch (Zend_Db_Adapter_Exception $e) {
+            $db = Adapter::factory($adapter, $dbInfos, $connect = false);
+
+            // database not found, we try to create  it
+            if ($db->isErrNo($e, '1049')) {
+                $dbInfosConnectOnly = $dbInfos;
+                $dbInfosConnectOnly['dbname'] = null;
+                @Db::createDatabaseObject($dbInfosConnectOnly);
+                @DbHelper::createDatabase($dbInfos['dbname']);
+
+                /* // select the newly created database */
+                @Db::createDatabaseObject($dbInfos);
+            } else {
+                throw $e;
+            }
+        }
+
+        @DbHelper::checkDatabaseVersion();
+
+        @Db::get()->checkClientVersion();
+        self::createConfigFile($dbInfos);
+        return $dbInfos;
+    }
+    
+    /**
+     * Installation Step 4: Table Creation
+     */
+    public static function tablesCreation()
+    {
+        //Could have already created tables ?
+        DbHelper::createTables();
+        DbHelper::createAnonymousUser();
+        DbHelper::recordInstallVersion();
+
+        self::updateComponents();
+
+        Updater::recordComponentSuccessfullyUpdated('core', Version::VERSION);
+    }
+
+    /**
+     * Installation Step 5: General Set-up (superuser login/password/email and subscriptions)
+     */
+    public static function setupSuperUser($settings)
```

Comment:
Any reason for having all methods static? Imho it would be better to have them non-static. Additionally I would set the settings as private property in `__construct` maybe and use that in all methods instead of passing it to every method

Reply:
I'm not a php developper, and I had some problem with the a php feature where you store initialized class, so I made it like that. Everything can be improve.